### ISSUE TITLE
Submenu wrong aria role

### ISSUE THREAD

**Issue by jbmolle** [tusen-ai/naive-ui#5729]
### Describe the bug

In lighthouse, it complains that menu items with role="menuitem" have wrong parent in submenus because submenu also has a role="menuitem".
According to https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role, submenus should have the role="menu" instead.
In src/menu/src/Submenu.tsx, lines 216 and 229, role="menuitem" should be replaced by role="menu"

### Steps to reproduce

1. Create a menu with submenu and menu items
2. Launch Lighthouse check
3. Error on the aria role

### Link to minimal reproduction

none

### System Info

```Shell
``
  System:
    OS: Linux 5.15 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    CPU: (4) x64 Intel(R) Core(TM) i5-4440 CPU @ 3.10GHz
    Memory: 6.60 GB / 15.30 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.5.1 - /usr/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 9.8.1 - /usr/bin/npm
    pnpm: 8.14.1 - /usr/bin/pnpm
    bun: 1.0.6 - /usr/bin/bun
  Browsers:
    Brave Browser: 122.1.63.162
    Chrome: 111.0.5563.110
    Chromium: 122.0.6261.69
  npmPackages:
    naive-ui: ^2.38.1 => 2.38.1
``
```

### Used Package Manager

yarn

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/tusen-ai/naive-ui/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://www.naiveui.com/en-US/).
- [X] Check that there isn't [already an issue](https://github.com/tusen-ai/naive-ui/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/tusen-ai/naive-ui/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.
